labview/docs/InstrumentStudio Plug-In Generator Guide.md

```diff
@@ -0,0 +1,160 @@
+[JKI VI Package Manager](https://www.ni.com/en/support/downloads/tools-network/download.jki-vi-package-manager.html#443251) | 2021 SP1 or later
+
+---
+
+## Installation
+
+Download the `InstrumentStudio Plug-In SDK` and `InstrumentStudio Plug-In Generator` packages from
+the latest release assets.
+
+**Note:** Please install the `InstrumentStudio Plug-In SDK` package before installing
+`InstrumentStudio Plug-In Generator` package.
+
+---
+
+## Developing a LabVIEW InstrumentStudio Plug-In
+
+1. Create and save a new LabVIEW project.
+2. From the project window, go to `Tools` → `Plug-In SDKs` → `InstrumentStudio Plug-In` → `Create
+   InstrumentStudio Plug-in...`.
+   1. This will open the `Create InstrumentStudio Plug-in` dialog.
+   2. The name of the Active LabVIEW Project will be automatically populated in the `Active Project
+      Name` indicator.  
+      **Note:** The new InstrumentStudio plug-in will be created in this active LabVIEW project.
+   3. Enter the `InstrumentStudio Plug-in Name` and `InstrumentStudio Plug-in Group`.
+   4. Click on the `Create InstrumentStudio Plug-in` button.
+
+      ![Create InstrumentStudio Plug-In](./images/InstrumentStudio%20Plug-In%20Generator%20Guide/Create%20InstrumentStudio%20Plug-In.png)
+
+   5. This will create a new InstrumentStudio plug-in library and a PPL build specification for the
+      plug-in in the Active LabVIEW project.
+
+      ![InstrumentStudio Plug-In Library](./images/InstrumentStudio%20Plug-In%20Generator%20Guide/InstrumentStudio%20Plug-In%20Library.png)
+
+3. By default, the top-level plug-in VI contains the template logic that demonstrates the basic
+   features of the LabVIEW InstrumentStudio plug-ins.
+
+   ![Top-level plug-in VI Block diagram](./images/InstrumentStudio%20Plug-In%20Generator%20Guide/Top-level%20VI%20Block%20diagram.png)
+
+4. Add your logic implementation to the top-level plug-in VI.
+      - For reference, please refer to the example plugins in the repository.
+
+**Note:** For more information about the GPluginData file format and InstrumentStudio plug-in
+development, please refer to the
+[GPluginData-File-Format](https://github.com/ni/instrumentstudio-plugins/blob/main/labview/docs/GPluginData-File-Format.pdf) and
+[G Plugin Development Guide](https://github.com/ni/instrumentstudio-plugins/blob/main/labview/docs/G%20Plugin%20Development%20Guide.pdf)
+documents.
+
+---
+
+## Deploying a LabVIEW InstrumentStudio plug-in
+
+1. Open the Active LabVIEW project where the InstrumentStudio plug-in was created.
+2. The generated plug-in comes with a Packed Project Library (PPL) build specification.
+3. To build the PPL, right-click on the desired build specification and choose 'Build'.
+4. Copy the built plug-in files into the InstrumentStudio `Addons` directory, which is `C:\Program
+   Files\National Instruments\InstrumentStudio\Addons` by default.
```

Comment:
You may also want to mention plugins can install to a user defined directory configured in IS preferences dialog for plugins. This directory is intended to facilitate installation of plugins without requiring administrative permissions. Configuring this preference requires IS to be launched with administrator privileges.

![image](https://github.com/user-attachments/assets/1bd0b3b9-9bc9-4800-a83b-c356e65a9ac6)